### PR TITLE
pythonPackages.humanfriendly: init at 4.18

### PR DESCRIPTION
**Pull request by montag451**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/humanfriendly/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    inherit pname version;
+    sha256 = "33ee8ceb63f1db61cce8b5c800c531e1a61023ac5488ccde2ba574a85be00a85";
+  };
+
+  # humanfriendly tests depends on coloredlogs which itself depends on
+  # humanfriendly. This lead to infinite recursion when trying to
+  # build this package so we have to disable the test suite :(
+  doCheck = false;
+
+  meta = with stdenv.lib; {
```
Comment (dotlambda) [inline pkgs/development/python-modules/humanfriendly/default.nix]:
```suggestion
  meta = with lib; {
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/humanfriendly/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    description = "Human friendly output for text interfaces using Python";
+    homepage = https://pypi.org/project/humanfriendly/;
```
Comment (dotlambda) [inline pkgs/development/python-modules/humanfriendly/default.nix]:
```suggestion
    homepage = https://humanfriendly.readthedocs.io/;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.humanfriendly python3.pkgs.humanfriendly

**Review by dotlambda — CHANGES_REQUESTED** [review]
> No matching distribution found for monotonic; python_version == "2.6" or python_version == "2.7" or python_version == "3.0" or python_version == "3.1" or python_version == "3.2" (from humanfriendly==4.18)

File: pkgs/development/python-modules/humanfriendly/default.nix (hunk 2 above)
Comment (montag451) [reply pkgs/development/python-modules/humanfriendly/default.nix]: See 767ddae15df11f266b69412ee45ff398cce5d24f

**Comment by montag451** [comment]
Build failure fixed in 984a3567aa11ab8e1e689e3ae4ef999134fd0eef

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.humanfriendly python3.pkgs.humanfriendly

File: pkgs/development/python-modules/humanfriendly/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+buildPythonPackage rec {
+  pname = "humanfriendly";
+  version = "4.18";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "33ee8ceb63f1db61cce8b5c800c531e1a61023ac5488ccde2ba574a85be00a85";
+  };
+
+  propagatedBuildInputs = [] ++ lib.optional (pythonOlder "3.3") monotonic;
```
Comment (dotlambda) [inline pkgs/development/python-modules/humanfriendly/default.nix]:
```suggestion
  propagatedBuildInputs = lib.optional (pythonOlder "3.3") monotonic;
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/humanfriendly/default.nix (hunk 3 above)
Comment (montag451) [reply pkgs/development/python-modules/humanfriendly/default.nix]: See 4aef43a7771b7d0f7bdd2ff4dfce67579f628f76

**Comment by dotlambda** [comment]
@GrahamcOfBorg build python2.pkgs.humanfriendly python3.pkgs.humanfriendly

**Pull request merged**